engine/src/main/java/org/camunda/bpm/engine/impl/form/handler/DefaultFormHandler.java

```diff
@@ -82,6 +90,36 @@ public void parseConfiguration(Element activityElement, DeploymentEntity deploym
       // provide support for new form field metadata
       parseFormData(bpmnParse, expressionManager, extensionElement);
     }
+
+    String formKeyAttribute = activityElement.attributeNS(BpmnParse.CAMUNDA_BPMN_EXTENSIONS_NS, "formKey");
+    String formRefAttribute = activityElement.attributeNS(BpmnParse.CAMUNDA_BPMN_EXTENSIONS_NS, "formRef");
+
+    if(formKeyAttribute != null && formRefAttribute != null) {
+      bpmnParse.addError("Invalid element definition: only one of the attributes formKey and formRef is allowed.", activityElement);
+    }
+
+    if (formKeyAttribute != null) {
+      this.formKey = expressionManager.createExpression(formKeyAttribute);
+    } else if(formRefAttribute != null) {
+      // formRef
+      this.camundaFormDefinitionKey = expressionManager.createExpression(formRefAttribute);
+      // formRefBinding
+      String formRefBindingAttribute = activityElement.attributeNS(BpmnParse.CAMUNDA_BPMN_EXTENSIONS_NS,
+          "formRefBinding");
+      List<String> allowedBindings = Arrays.asList("deployment", "latest", "version");
```

Comment:
Maybe it would be a good idea to convert the allowed values to `enum` or at least to static values in this class?

For example, the `version` value is also used on line 117. Furthermore, if we ever decide to add more allowed values (like `versionTag`), it would be better if they are easier to find.

Reply:
There is already a nested enum in `BaseCallableElement` called `CallableElementBinding` which includes all allowed values plus `versionTag`. I am sure we could reuse it when we decide to support `versionTag`. For now, I would say let's make them constants. Do you agree?

Reply:
Yes, constants are fine by me. :)